src/messaging/ExchangeMgr.cpp

```diff
@@ -203,23 +245,44 @@ void ExchangeManager::DispatchMessage(const PacketHeader & packetHeader, const P
             }
         }
     }
-    // Discard the message if it isn't marked as being sent by an initiator.
+    // Discard the message if it isn't marked as being sent by an initiator and the message does not needs to send
+    // ack to the peer.
     else
     {
-        ExitNow(err = CHIP_ERROR_UNSOLICITED_MSG_NO_ORIGINATOR);
+        if (!payloadHeader.IsAckMsg())
```

Comment:
nit: if you combine this as  `else if (!payloadHeader.IsAckMsg())` you can reduce one level of indent.

Reply:
Acked